app/src/main/java/com/infomaniak/mail/data/MailData.kt

```diff
@@ -366,22 +374,29 @@ object MailData {
 
         // Save new data
         Log.d("API", "Threads: Save new data")
-        folder.threads = apiThreads.toRealmList()
-        // TODO: Put this back (and make it work) when we have EmbeddedObjects
-        // folder.threads = apiThreads.map { apiThread ->
-        //     realmThreads?.find { it.uid == apiThread.uid }?.also { realmThread ->
-        //         apiThread.messages.forEach { apiMessage ->
-        //             realmThread.messages.find { it.uid == apiMessage.uid }?.also { realmMessage ->
-        //                 apiMessage.apply {
-        //                     fullyDownloaded = realmMessage.fullyDownloaded
-        //                     body = realmMessage.body
-        //                     attachments = realmMessage.attachments
-        //                 }
-        //             }
-        //         }
-        //     } ?: apiThread
-        // }.toRealmList()
-        MailboxContentController.upsertFolder(folder)
+        MailRealm.mailboxContent.writeBlocking {
+            apiThreads.forEach { apiThread ->
+                realmThreads?.find { it.uid == apiThread.uid }?.let { realmThread ->
+                    apiThread.messages.forEach { apiMessage ->
+                        realmThread.messages.find { it.uid == apiMessage.uid }
+                            ?.let { getLatestMessage(it.uid) }
```

Comment:
You can simply use `getLatestMessage` without make this `realmThread.messages.find { it.uid == apiMessage.uid }`